### DISCUSSION TITLE
Only joyful numbers?

### DISCUSSION THREAD

**Discussion by jrmadsen67** [AndrewFeeney/all-positive-integers-in-ascending-order#6] [Q&A]
when you say, "positive" you're referring to numbers that bring me joy, right?

so I can leave out ones like 17?

**Comment by AndrewFeeney** [comment]
If we were to adopt this approach, it would require a change to the contribution guidelines to drop the 'consecutive' requirement. Personally I feel that this would significantly reduce the utility of this library, and begs other questions.

As of right now, we're not up to 17 yet, so I propose we defer any such decisions until then. The next number which will be accepted is 2.

**Comment by AndrewFeeney** [comment]
Just reopening this as we've received a pull request that includes the number 17. Personally I'm in favour of continuing with the consecutive approach, since joyfulness is subjective. I don't personally have any problems with 17 whereas 43 can go eat rocks as far as I'm concerned.

**Reply by deplicator** [reply]
For reference #34 is the PR that brings in `17`.

At `16` we have a balance of even and odd numbers--8 each. I can understand the hesitance to add `17` as is breaks this balance (9 odd and 8 even). Presumably, whatever comes next, will bring back the balance... it's happened 8 times before and a pattern is established (#12, #14, #16, #20, #24, #28, #32).

We keep the emotion out of it and stay subjective this project will shape up to be useful for all!

**Comment by ellipsisComma** [comment]
For me the advantage of this library is definitely that it's fully consecutive—when I import it, `allPositiveIntegersInAscendingOrder[7]` *always* returns `8`, `allPositiveIntegersInAscendingOrder[allPositiveIntegersInAscendingOrder[7]]` *always* returns `9`, etc. It makes a complicated situation very simple.

**Reply by AndrewFeeney** [reply]
I think that's a great point. It really does help when using this library if we can assume that we can always find the `n`th integer by simply by looking up `allPositiveIntegersInAscendingOrder[n - 1]`. I think that settles it.
